clang/lib/Sema/SemaBounds.cpp

```diff
@@ -2081,19 +2442,30 @@ namespace {
                               Cast->getType()->isCheckedPointerPtrType());
       ProofStmtKind Kind = IsStaticPtrCast ? ProofStmtKind::StaticBoundsCast :
                              ProofStmtKind::BoundsDeclaration;
-      ProofResult Result =
-        ProveBoundsDeclValidity(TargetBounds, SrcBounds, Cause, nullptr, Kind);
+      FreeVariableListTy FreeVars;
+      ProofResult Result = ProveBoundsDeclValidity(
+          TargetBounds, SrcBounds, Cause, nullptr, FreeVars, Kind);
       if (Result != ProofResult::True) {
-        unsigned DiagId = (Result == ProofResult::False) ?
-          diag::error_static_cast_bounds_invalid :
-          (CSS != CheckedScopeSpecifier::CSS_Unchecked ?
-           diag::warn_checked_scopestatic_cast_bounds_invalid :
-           diag::warn_static_cast_bounds_invalid);
+        // Which diagnostic message to print?
+        unsigned DiagId =
+            (Result == ProofResult::False)
```

Comment:
Upon reflection, I actually don't think it makes sense to print an error for free variables in static pointer casts. You can just print the existing `error_static_cast_bounds_invalid` error message instead.